src/main/java/school/lemon/changerequest/java/introduction/hw2/Task10.java

```diff
@@ -8,7 +8,7 @@ public static String trim(String text) {
         if (text == null) {
             return null;
         }
-        String reg = "(((?<=[\\t]\\s)[A-Za-z]+.*[^\\s\\t]+)|([A-Z]).*[^\\t\\s])";
+        String reg = "(((?<=\\s)[A-Za-z0-9]+.*[^\\s]+)|([A-Z0-9]).*[^\\s])";
```

Comment:
And now `[A-Za-z0-9]` could be replaced with `\\w`.
Also, your next char class `[A-Z0-9]` is quite weird, there couldn't be small characters ?
I thought I've already mentioned, that group `[^\\s]` could be changed to `\\S`